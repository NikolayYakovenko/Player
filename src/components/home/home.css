.wrapper {
    min-height: 500px;
    max-width: 600px;
    margin: 30px auto;
    padding: 15px;
    border: 1px solid #d0d4dc;
    border-radius: 3px;
}