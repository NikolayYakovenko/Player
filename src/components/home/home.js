import React from 'react';
import PropTypes from 'prop-types';

import './home.css';


export const Home = (props) => {
    return (
        <div className='wrapper'>
            {props.children}
        </div>
    );
};

Home.propTypes = {
    children: PropTypes.any.isRequired,
};
