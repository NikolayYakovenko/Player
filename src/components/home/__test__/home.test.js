import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Home } from '../home';


describe('Check Home component', () => {
    const props = {
        children: <div>First test</div>,
    };

    test('should be selectable by class "wrapper"', () => {
        const wrapper = shallow(<Home {...props} />);
        expect(wrapper.is('.wrapper')).toBe(true);
    });

    test('should render without throwing an error', () => {
        const wrapper = shallow(<Home {...props} />);
        expect(wrapper
            .contains(<div className='wrapper'><div>First test</div></div>))
            .toBe(true);
    });

    test('renders children when passed in', () => {
        const wrapper = shallow(<Home {...props}><p>Hello</p></Home>);
        expect(wrapper.contains(<p>Hello</p>)).toBe(true);
    });

    test('renders correctly', () => {
        const rendered = renderer.create(<Home {...props} />);
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});

