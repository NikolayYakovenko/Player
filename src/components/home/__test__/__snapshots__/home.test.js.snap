// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Home component renders correctly 1`] = `
<div
  className="wrapper"
>
  <div>
    First test
  </div>
</div>
`;
